src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryDate.cs

```diff
@@ -481,6 +614,13 @@ private static FormulaValue DateParse(EvalVisitor runner, EvalVisitorContext con
 
             if (DateTime.TryParse(str, runner.CultureInfo, DateTimeStyles.None, out var result))
             {
+                var tzi = runner.GetService<TimeZoneInfo>() ?? TimeZoneInfo.Local;
+
+                if (result.Kind == DateTimeKind.Local)
+                {
+                    result = TimeZoneInfo.ConvertTime(result, TimeZoneInfo.Local, tzi);
```

Comment:
>result = TimeZoneInfo.ConvertTime(result, TimeZoneInfo.Local, tzi); [](http://example.com/codeflow?start=20&length=67)

To test this conversion it would be good to have tests for Date[Time]Parse with values in different time zones and unspecified (maybe we already have those, but we should check):

- Date[Time]Value("2022-09-29T20:00:00")
- Date[Time]Value("2022-09-29T20:00:00Z")
- Date[Time]Value("2022-09-29T20:00:00-08:00")

Reply:
There were already some tests in place but I've added those as well